### ISSUE TITLE
[raspberry] no way to disable password login via ssh

### ISSUE THREAD

**Issue by hagabla** [plexinc/plex-media-player#518]
"passwd" is disabled in the installation and suggests to disable ssh password authentication.
Due to squashfs is not possible to edit sshd_config. There is no way to disable password auth in the app, as it is in libreelec/kodi.

Everyone can login as root on the PMP which is obviously problematic.

`PlexMediaPlayer:~ # passwd

 There is no working 'passwd'.

 The 'passwd' command changes passwords for user accounts.

 With LibreELEC it is not possible to change the system password

 SSH is included only as a last support resort. SSH is off by default.
 Most users never need SSH and need help using it so we need a default
 password. If you need to keep SSH always on then this is unsupported
 but can be secured with certificates.

 TIP: disable password authentication in ssh and use public key authentication.`

**Comment by LongChair** [comment]
This is more a LibreELEC feature which we use. maybe that request should be made to LE so that we can benefit from it. We try to keep the changes from Upstream LIbreELEC miniaml so that we can follow them easily.

**Comment by hagabla** [comment]
Well, Libreelec has a way of doing this in their GUI. It is baked in in their Kodi. So raising an issue there does not help, it's an Plex issue (imho)

**Comment by LongChair** [comment]
I don't agree that most users don't need SSH at this point.

There is no other way to connect with Wifi than going into SSH. Same thing goes for Setting Timezone.
Also most of the users don't have a techy background and would probably have trouble to setup a public key Auth.
Additionnally i don't really see the point of hacking an Embeeded device as most of the system is readonly due to squashfs.

In the principle i agree that having a root/plex identification is not good in terms of security (should security really matter on such a device), but we will need to provide an easy way to setup this if we need to and some other essential settings like the ones i mentionned should not require ssh anymore.

**Comment by bqq100** [comment]
This really does need to get addressed.  In particular a default root password that can't be changed at all and SSH enabled by default, that raises a big red flag.  

As to the point of hacking an embedded device, they can be used by botnets to launch DDOS attacks at a minimum (see Mirai and hacked IP cams).

**Comment by wm4** [comment]
I agree. We planned to disable it by default and to add UI for this.

I wonder if there's some cleverer way for the default password... maybe we could generate a random one and display it at boot or something? (The idea is that only people who have access anyway see it.)

**Comment by moshegottlieb** [comment]
Having a root password enabled linux with a known password running in my house is a big **no no**.  
Here's a temporary workaround until this is fixed: 

1. Copy a public key to the media player
1. Add it to the authorized keys
1. Copy the `sshd` config file to a writable location
1. Modify it to disable passwords
1. Kill `sshd` in `autostart.sh` and restart it with the modified config file

Example:

```
$ # On my local machine
$ PLEX_IP = "enter the IP address of the plex media player here instead of this string"
$ scp ~/.ssh/id_rsa.pub root@${PLEX_IP}
$ ssh root@${PLEX_IP}
PlexMediaPlayer:~ # cat id_rsa.pub >> .ssh/authorized_keys
PlexMediaPlayer:~ # rm id_rsa.pub
PlexMediaPlayer:~ # mkdir -p .local/etc/ssh
PlexMediaPlayer:~ # cp /etc/ssh/sshd_config .local/etc/ssh/
PlexMediaPlayer:~ # vi .local/etc/ssh/sshd_config
PlexMediaPlayer:~ # # Uncomment : PasswordAuthentication yes 
PlexMediaPlayer:~ # # and change to : PasswordAuthentication no
PlexMediaPlayer:~ # echo "killall sshd && /usr/sbin/sshd -f /storage/.local/etc/ssh/sshd_config" >> .config/autostart.sh
PlexMediaPlayer:~ # chmod +x  .config/autostart.sh # In case you've just created it
```

Enjoy!
